src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/notifications/CaseNotesNotificationsService.kt

```diff
@@ -22,15 +24,24 @@ class CaseNotesNotificationsService(
   private val hmppsAuthService: HMPPSAuthService,
   private val userTypeChecker: UserTypeChecker,
   private val communityAPIOffenderService: CommunityAPIOffenderService,
+  private val caseNoteRepository: CaseNoteRepository,
 ) : ApplicationListener<CaseNoteEvent>, NotifyService {
   companion object : KLogging()
 
   @AsyncEventExceptionHandling
   override fun onApplicationEvent(event: CaseNoteEvent) {
     when (event.type) {
       CaseNoteEventType.SENT -> {
-        emailAssignedCaseWorker(event.caseNote)
-        emailResponsibleProbationPractitioner(event.caseNote)
+        // get case note from repository because the linking objects (referral.currentAssignee) on the caseNote from the event
+        // may sometimes not be lazy loaded because the session may already have been closed by this point
+       caseNoteRepository
```

Comment:
Can we use repositories here?

Reply:
should not be a problem because we're not performing any write operations - but I've changed it to a service to be more consistent with how we encapsulate Repository methods.